W-21215932-Update-Mule-runtime-content-CP (#583)
mulesoft/docs-code-builder: modules/ROOT/pages/troubleshoot-mule-runtime-errors.adoc

include::reuse::partial$beta-banner.adoc[tag="anypoint-code-builder"]

> Anypoint Code Builder includes bundled Mule runtime versions that enable instant project opening and design. When you run or debug applications, the IDE checks for the exact runtime version specified in your project. If the exact version isn't available locally, you'll see error dialogs that help you resolve the issue.

Reviewer: ```suggestion
Anypoint Code Builder includes bundled Mule runtime versions that enable instant project opening and design. When you run or debug applications, the IDE checks for the exact runtime version specified in your project. If the exact version isn't available locally, error dialogs appear to help you resolve the issue.
```

Author: OK! Thanks!

Don't change the default locations of Mule. If you receive an error that Mule isn't found or installed, configure the Mule Home path.

> If configuring your Mule runtime engine, ensure you're defining the `mule.homeDirectory` property. 

Reviewer: ```suggestion
When you configure Mule runtime, ensure you define the `mule.homeDirectory` property. 
```

Author: OK! Thanks!

== Mule Runtime Version Not Defined

The error *Mule runtime version not defined* occurs when your project doesn't specify a Mule runtime version in the project's `mule-artifact.json` file.

To resolve this issue:

. Click *Open Project Properties* in the error dialog, or run the command `MuleSoft: Open Mule Project Properties`.
. In *Project Properties*, select a Mule runtime version.
. Click *Apply*.

== Java Version Not Defined

The error *Java version not defined* occurs when your project doesn't specify a Java version in the project's `mule-artifact.json` file.

To resolve this issue:

. Click *Open Project Properties* in the error dialog, or run the command `MuleSoft: Open Mule Project Properties`.
. In *Project Properties*, select a Java version.
. Click *Apply*.

== Mule Runtime Version Not Found (Compatible Version Available)

The error *Mule runtime [version] not found* occurs when you try to run or debug an application, but the exact Mule runtime version specified in your project isn't available locally. However, a compatible local version is available.

To resolve this issue, you have two options:

> * **Continue with compatible version**: Click *Continue* to use the latest local compatible version. The dialog shows which version will be used.

Reviewer: ```suggestion
* **Continue with compatible version**: Click *Continue* to use the latest local compatible version. The dialog shows which version to use.
```

Author: OK! Thanks!
